saleor/checkout/fetch.py

```diff
@@ -290,95 +284,85 @@ def fetch_checkout_info(
     lines: Iterable[CheckoutLineInfo],
     discounts: Iterable["DiscountInfo"],
     manager: "PluginsManager",
-    *,
-    # Ugly hack to disable fetching shipping methods if they aren't needed
-    # Preventing multiple redundant and potentially costful API calls in
-    # checkout lines add/update/delete mutations
-    fetch_delivery_methods: bool = True,
+    shipping_channel_listings: Optional[
```

Comment:
I'm 50:50 on this one. Shipping method prices do not change in the mutations, but sometimes need to be used twice in a single request, e.g. when adding a promo code to a checkout:

```
shipping_channel_listings = checkout.channel.shipping_method_listings.all()
checkout_info = fetch_checkout_info(
    checkout, lines, discounts, manager, shipping_channel_listings
)

add_promo_code_to_checkout(
    manager,
    checkout_info,
    lines,
    promo_code,
    discounts,
)

update_delivery_method_lists_for_checkout_info(
    checkout_info,
    checkout_info.checkout.shipping_method,
    checkout_info.checkout.collection_point,
    checkout_info.shipping_address,
    lines,
    discounts,
    manager,
    shipping_channel_listings,
)
```

Calling prefetch_related on channel would be enough to avoid double database calls, but passing it as an argument, at least to me, makes it more obvious